src/Symfony/Component/Form/Extension/Templating/Type/FormTypeThemeExtension.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Symfony\Component\Form\Extension\Templating\Type;
+
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormRendererEngineInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Damian Wróblewski <damianwroblewski75@gmail.com>
+ * Allows to define from theme as form type option
+ * The form them is passed to rendering engine when Form::createView() is called.
+ */
+class FormTypeThemeExtension extends AbstractTypeExtension
+{
+    /**
+     * @var FormRendererEngineInterface
+     */
+    private $rendererEngine;
+
+    /**
+     * @var string
+     */
+    private $extension;
+
+    public function __construct(FormRendererEngineInterface $rendererEngine, $extension)
+    {
+        $this->extension = $extension;
+        $this->rendererEngine = $rendererEngine;
+    }
+
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver
+            ->setDefault('theme', null)
+            ->setDefault('theme_extension', $this->extension);
```

Comment:
@ogizanagi Why not sure? This pull request is actually all about this option. It would be fully backwards compatible anyway.

Reply:
You already pass the extension in the constructor. I'm not sure about exposing the `theme_extension` option then, and I don't really think it should be changed.
BTW, what happen when you have both php and twig templating engines registered? This form type extension will be registered twice, right (one for each engine)? Depending on the order they are called, the `theme_extension` default will change unless I'm missing something.

Reply:
Oh you mean the file_extension. Sorry didn't get it.
I wanted to account the fact that someone may have some non-standard file extensions. If extension is left null then it is automatically defined basing on rendering engine (php/twig). But you can also force the extension yourself in case your template file has non-stanrad extension for any reason. 

Someone may use: `file_name.twig` format. In that case he can simply change the `theme_extension` to force `.twig` file extension.